### DISCUSSION TITLE
Refactoring steps from the CLI?

### DISCUSSION THREAD

**Discussion by EGTomKies** [getgauge/gauge#2336] [General]
I was curious, is there still a CLI way to refactor steps?  In an earlier version of the documentation there was a `--refactor` flag, but it doesn't seem to still be available.  

If it's actually gone, what's the preferred method of refactoring a step using python?

**Comment by EGTomKies** [comment]
I pulled the latest getgauge-python and can see there's a refactor.py in there, so it's likely still an option.  If someone has a pointer to documentation on usage, that'd be welcome, thanks.

**Reply by zabil** [reply]
The cli refactor command  is no longer available in newer versions of gauge. It was removed as it was difficult to maintain changes and keep it in sync with the ui version.

**Reply by EGTomKies** [reply]
That's unfortunate, thank you.  You mention a UI version, is there any information on that?  


I'm still in need of a best practice for updating some steps.

**Reply by zabil** [reply]
If you use, VS code you can use the gauge vs code plugin https://marketplace.visualstudio.com/items?itemName=getgauge.gauge to refactor.

**Reply by EGTomKies** [reply]
Fantastic, thank you!

**Reply by EGTomKies** [reply]
Sorry to be a bit daft, but when I try selecting a step and then "Refactor" from the context menu, it's telling me "No refactorings available".   

I'm sure I'm missing something simple.  It's version 0.1.0 of the plug in, and I just updated VSCode yesterday to latest

**Reply by zabil** [reply]
You should place your cursor on the step you want to refactor and then press F2 to rename a function

**Reply by EGTomKies** [reply]
The problem is that we're not looking to rename the underlying function, but rather the _step_ that is used to call it.  

So rather than: 

`Click the element with automation id: <automation_id>`

we might use: 

`Click the element by automation id <automation_id> to do xyz`

From what I understand, the old CLI function would have allowed such a change, but I'm having an issue in sorting out how we'll do it now.
